### PR TITLE
Move conversion of minimum index compatibility to IndexVersion

### PR DESCRIPTION
**Pull request by thecoop**
Move final calls to convert index compatibility from Version to IndexVersion

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/Build.java [hunk 1]
```diff
@@ -115,12 +116,22 @@ private static Build findLocalBuild() {
...
         String minWireCompat = Version.CURRENT.minimumCompatibilityVersion().toString();
-        String minIndexCompat = Version.CURRENT.minimumIndexCompatibilityVersion().toString();
+        String minIndexCompat = minimumCompatString(IndexVersion.MINIMUM_COMPATIBLE);
         String displayString = defaultDisplayString(type, hash, date, version);
 
         return new Build(flavor, type, hash, date, isSnapshot, version, minWireCompat, minIndexCompat, displayString);
     }
 
+    public static String minimumCompatString(IndexVersion minimumCompatible) {
+        if (minimumCompatible.before(IndexVersion.V_8_11_0)) {
```
Comment (ldematte) [inline server/src/main/java/org/elasticsearch/Build.java:126]: Shouldn't this be 8.10?

Comment (thecoop) [reply server/src/main/java/org/elasticsearch/Build.java:126]: Its `before`, not `onOrBefore`, so its exclusive of 8.11

**Comment by thecoop** [comment]
@elasticsearchmachine rerun elasticsearch-ci/bwc

**Review by rjernst — APPROVED** [review]
LGTM

**Pull request merged**